sunpy/image/coalignment.py

```diff
@@ -342,6 +338,58 @@ def repair_image_nonfinite(image):
     return repaired_image
 
 
+@quantity_input(yshift=u.pix, xshift=u.pix)
+def apply_shifts(mc, yshift, xshift, clip=True):
+    """Apply a set of pixel shifts to a mapcube, and return a new mapcube.
+
+    Parameters
+    ----------
+    mc : sunpy.map.MapCube
+        A mapcube of shape (ny, nx, nt), where nt is the number of layers in
+        the mapcube.  'ny' is the number of pixels in the y direction, 'nx'
+        is the number of pixels in the 'x' direction.
+
+    yshift : `~astropy.units.Quantity` instance
+        An array of pixel shifts in the y-direction for an image.
+
+    xshift : `~astropy.units.Quantity` instance
+        An array of pixel shifts in the x-direction for an image.
+
+    clip : bool
+        If True, then clip off x, y edges in the datacube that are potentially
+        affected by edges effects.
+
+    Returns
+    -------
+    sunpy.map.MapCube
+        A mapcube of the same shape as the input.  All layers in the mapcube
+        have been shifted according the input shifts.
+    """
+
+    # new mapcube will be constructed from this list
+    newmc_list = []
+
```

Comment:
Would it not be worth validating that all the maps in the mapcube are the same shape?